ddtrace/contrib/django/patch.py

```diff
@@ -666,14 +674,187 @@ def django_asgi_modifier(span, scope):
     return TraceMiddleware(func(*args, **kwargs), integration_config=config.django, span_modifier=django_asgi_modifier)
 
 
+_POSSIBLE_USER_ID_FIELDS = ["pk", "id", "uid", "userid", "user_id", "PK", "ID", "UID", "USERID"]
+_POSSIBLE_LOGIN_FIELDS = ["username", "user", "login", "USERNAME", "USER", "LOGIN"]
+_POSSIBLE_EMAIL_FIELDS = ["email", "mail", "address", "EMAIL", "MAIL", "ADDRESS"]
+_POSSIBLE_NAME_FIELDS = ["name", "fullname", "full_name", "NAME", "FULLNAME", "FULL_NAME"]
+
+
+def _find_in_user_model(user, possible_fields):
+    for field in possible_fields:
+        value = getattr(user, field, None)
+        if value:
+            return value
+
+    return None  # explicit to make clear it has a meaning
+
+
+def _get_userid(user):
+    user_login = getattr(user, config._user_model_login_field, None)
+    if user_login:
+        return user_login
+
+    return _find_in_user_model(user, _POSSIBLE_USER_ID_FIELDS)
+
+
+def _get_username(user):
+    username = getattr(user, config._user_model_name_field, None)
+    if username:
+        return username
+
+    if hasattr(user, "get_username"):
+        return user.get_username()
```

Comment:
Since we're running an unknown function, should we wrap this in a try except? Not sure if this is being overly cautious. It looks like we do this in some other places too e.g. `_get_name()`

Reply:
Technically I guess we don't even know if it's a function right? Or that it returns a string?

Reply:
It's part of the base User model of django (as a method) but you're right that it could be anything else on user code. All the login tracing stuff is wrapped in a try... except (with logging) in `traced_login` so I think that could cover these corner cases pretty well however in this case I think it merits an additional check to continue trying to extract the username (same for `get_full_name`). Fixed on next commit.